src/system/SettingsModule/Api/LocaleApi.php

```diff
@@ -30,12 +32,19 @@ class LocaleApi implements LocaleApiInterface
     private $kernel;
 
     /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
      * LocaleApi constructor.
      * @param ZikulaHttpKernelInterface $kernel
+     * @param RequestStack $requestStack
      */
-    public function __construct(ZikulaHttpKernelInterface $kernel)
+    public function __construct(ZikulaHttpKernelInterface $kernel, RequestStack $requestStack)
     {
         $this->kernel = $kernel;
+        $this->request = $requestStack->getCurrentRequest();
```

Comment:
better to delay acquiring the CurrentRequest until within the actual method and not store as class prop